bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdataframe.py

```diff
@@ -177,6 +177,28 @@ def pypowarray(numpyvec, pow):
 from ._pyz_utils import MethodTemplateGetter, MethodTemplateWrapper
 
 
+# For refernces to keep alive the NumPy arrays that are read by
+# MakeNumpyDataFrame.
+_numpy_data = {}
+
+
+def _numpy_data_add_ref(data):
+    _numpy_data[id(data)] = data
```

Comment:
Is this extra function necessary like the deleter function? Or can be just done inline in `RDataFrameAsNumpy`?

Reply:
No, that's not necessary. I will remove it.